k8s/pkg/apis/longhorn/v1beta2/backuptarget.go

```diff
@@ -71,3 +71,7 @@ type BackupTargetList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []BackupTarget `json:"items"`
 }
+
+// Hub defines the current version (v1beta2) is the storage version
+// so mark this as Hub
+func (bt *BackupTarget) Hub() {}
```

Comment:
So it means we need to change the hub interface implementation to the next version of resources from v1beta2 if we have v1beta3 for example.

Reply:
Once `v1beta3` introduce and it's the hub, and both v1beta1 and v1beta2 keeps, we need to change the hub interface implementation under `v1beta3` package, and add conversion from v1beta1 >-> v1beta3 in the v1beta1 package, and v1beta2 <-> v1beta3 in the v1beta2 package as well.

If only v1beta2 keeps and v1beta1 drop, we only need to add conversion from v1beta2 <-> v1beta3 in the v1beta2 package.

Reply:
thanks, also as the pattern we probably will adopt is just keep one convertible version like you mentioned before.